go/app/main.go

```diff
@@ -13,43 +13,60 @@ import (
 )
 
 const (
-	ImgDir = "images"
+	imgDir = "images"
+	port   = "9000"
 )
 
 type Response struct {
 	Message string `json:"message"`
 }
 
-func root(c echo.Context) error {
-	res := Response{Message: "Hello, world!"}
+// hello is an endpoint to return a Hello, world! message.
+func hello(c echo.Context) error {
+	res := Response{
+		Message: "Hello, world!",
+	}
 	return c.JSON(http.StatusOK, res)
 }
 
+// addItem is an endpoint to add a new item.
 func addItem(c echo.Context) error {
 	// Get form data
 	name := c.FormValue("name")
 	c.Logger().Infof("Receive item: %s", name)
 
 	message := fmt.Sprintf("item received: %s", name)
-	res := Response{Message: message}
+	res := Response{
+		Message: message,
+	}
 
-	// http.StatusCreated(201) is also good choice.StatusOK
-  // but in that case, you need to implement and return a URL
-  //   that returns information on the posted item.
+	// http.StatusCreated(201) is also acceptable.
 	return c.JSON(http.StatusOK, res)
 }
 
-func getImg(c echo.Context) error {
-	// Create image path
-	imgPath := path.Join(ImgDir, c.Param("imageFilename"))
+// getImage is an endpoint to return an image.
+func getImage(c echo.Context) error {
+	// Build image path
+	imgPath := filepath.Join(imgDir, filepath.Clean(c.Param("imageFilename")))
+	rel, err := filepath.Rel(imgDir, imgPath)
+	if err != nil || strings.HasPrefix(rel, "..") {
+		res := Response{
+			Message: "Invalid image path",
+		}
+		return c.JSON(http.StatusBadRequest, res)
+	}
```

Comment:
I've added to avoid path traversal. The previous code can access the outside of the code.

PoC:
```bash
$ curl --path-as-is 'localhost:9000/image/../../python/images/default.jpg'
```